charts/spire/charts/spire-server/templates/controller-manager-cluster-ids.yaml

```diff
@@ -1,42 +1,72 @@
 {{- $root := . }}
-{{- with .Values.controllerManager }}
-{{- if and (eq (.enabled | toString) "true") (eq (.identities.enabled | toString) "true") }}
+{{- range $key, $value := .Values.controllerManager.identities.clusterSPIFFEIDs }}
+{{-   range $skey, $svalue := $value }}
+{{-     if not (has $skey (list "name" "annotations" "labels" "enabled" "admin" "dnsNameTemplates" "downstream" "federatesWith" "jwtTTL" "namespaceSelector" "podSelector" "spiffeIDTemplate" "ttl" "workloadSelectorTemplates" "autoPopulateDNSNames")) }}
+{{-       fail (printf "Unsupported property specified: %s" $skey) }}
+{{-     end }}
+{{-   end }}
+{{-   range $rprop := list "spiffeIDTemplate" }}
+{{-     if not (hasKey $value $rprop) }}
+{{-       fail (printf "Required property %s was not specified" $rprop) }}
+{{-     end }}
+{{-   end }}
+{{-   if eq ($root.Values.controllerManager.enabled | toString) "true" }}
+{{-     if or (not (hasKey $value "enabled")) (eq ($value.enabled | toString) "true") }}
+---
 apiVersion: spire.spiffe.io/v1alpha1
 kind: ClusterSPIFFEID
 metadata:
-  name: {{ $root.Release.Namespace }}-{{ include "spire-controller-manager.fullname" $root }}-service-account-based
-  namespace: {{ include "spire-server.namespace" $root }}
+  {{- if $value.name }}
+  name: {{ $value.name }}
+  {{- else }}
+  name: {{ $root.Release.Namespace }}-{{ $root.Release.Name }}-{{ $key }}
+  {{- end }}
```

Comment:
This new functionality to override the name is not captured in docs. Is that intentional?

Reply:
Kind of unintentional. Not sure how to get it into the docs without setting it. If set, people might think they need to set it to something or set it to blank, where leaving it out is ok. And I think in the vast majority of cases, people wont need it.

Reply:
I think I could be convinced we don't need the .name override at all. But felt it didn't hurt anything just in case.

Reply:
if there is a use case for it, it should be documented. Otherwise I suggest to remove it.

Reply:
Only thing I could think about is if you install multiple instances in the same namespace. Then the release name is required to isolate the instances.